### PR TITLE
Decals/distance culling

### PR DESCRIPTION
**Pull request by Paulm-Unity**

### CODE REVIEW

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipelineAsset.asset [hunk 1]
```diff
@@ -3,7 +3,7 @@
 --- !u!114 &11400000
 MonoBehaviour:
   m_ObjectHideFlags: 0
-  m_PrefabParentObject: {fileID: 0}
+  m_CorrespondingSourceObject: {fileID: 0}
```
Comment (Paulm-Unity) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipelineAsset.asset]: I don't know what this is or why it changed

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Decal/DecalSystem.cs [hunk 2]
```diff
@@ -152,19 +179,27 @@ public void EndCull()
...
-                    instanceCount++;
-                    if (instanceCount == kDrawIndexedBatchSize)
-                    {
-                        m_PropertyBlock.SetMatrixArray(m_NormalToWorldID, m_InstanceNormalToWorld);
-                        cmd.DrawMeshInstanced(m_DecalMesh, 0, KeyMaterial, 0, m_InstanceMatrices, kDrawIndexedBatchSize, m_PropertyBlock);
-                        instanceCount = 0;
-                    }
+					// do additional culling based on individual decal draw distances
+					float distanceToDecal = (cameraPos - m_BoundingSpheres[decalIndex].position).magnitude;
+					float cullDistance = m_CachedDrawDistances[decalIndex].x + m_BoundingSpheres[decalIndex].radius;
```
Comment (Paulm-Unity) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Decal/DecalSystem.cs]: Still need to test this for performance

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassDBuffer.hlsl [hunk 3]
```diff
@@ -31,24 +31,14 @@ void Frag(  PackedVaryingsToPS packedInput,
...
-#endif
-#if _NORMALMAP
-		mask |= DBUFFERHTILEBIT_NORMAL;
-#endif
-#if _MASKMAP
-		mask |= DBUFFERHTILEBIT_MASK;
-#endif
 		uint oldVal = UnpackByte(_DecalHTile[posInput.positionSS.xy / 8]);
-		oldVal |= mask;
+		oldVal |= surfaceData.HTileMask;
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassDBuffer.hlsl]: Sorry , similar remark than before :), you don't have the right to access surfaceData here, what I was suggesting was that GetSurfaceData(positionDS.xz, decalToWorld, surfaceData); return the mask. As TileMask is "core feature of the decal, it is fine to define our own prototype for it.

Mean prototype of surface decal is  uint GetSurfaceData(pos, maxtrix4x4 SurfaceData);

I can update this later, no worry

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipelineAsset.asset (hunk 1 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDRenderPipelineAsset.asset]: ok, don't know either :). Let's see

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Decal/DecalSystem.cs (hunk 2 above)
Comment (sebastienlagarde) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Decal/DecalSystem.cs]: ok, I will still merge the PR, but consider testing :)

**Pull request merged**